azure-mgmt-compute/src/main/java/com/azure/management/compute/implementation/VirtualMachineScaleSetImpl.java

```diff
@@ -1468,14 +1468,15 @@ public VirtualMachineScaleSetImpl withoutUserAssignedManagedServiceIdentity(Stri
     // Create Update specific methods
     //
     @Override
-    protected void beforeCreating() {
+    protected Mono<Void> beforeCreating() {
```

Comment:
Why we are having such kind of change from void to Mono<Void>?

Reply:
Because this function could be implemented with blocking, which is not allowed during creation, so I change to Mono which can be chained.

Reply:
But in that case, you could wrap in createResourceAsync when it's required. I doubt if it is good style to do like this. After your changes, a lot of code snippet will work as it is while it just returns Mono.empty().

Reply:
Yes, but I'm afraid something bad happened if I change the blocking into `createResourceAsync`. The framework is not written by me, I can't guarantee to change everywhere. Anyway, I could try to change it. And actually, for this implementation, `createResourceAsync` is not necessary, it is splited into `beforeCreating`, `createInner` and `afterCreating` three necessary function. It could be strange when implement them all.

Reply:
I am not sure but from v1 code in `GroupableParentResourceImpl` it seems the intention for beforeCreating and afterCreating is do things no blocking (so no request). But we have bug in v1 that do block.

So it fails in v2 after migration.

Agree it might be better for us to fix the blocking issue in before/after if possible, as done in latest commit.

Reply:
Change it back, and use non-blocking chaining in another way.